src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+    constructor (name, health, strength){
+        super(health, strength);
+        this.name = name;
+    }
+    
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        } else return `${this.name} has died in act of combat`;
+    }
+
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`;
+        } else return `A Saxon has died in combat`;
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+    addViking(Viking){
+        this.vikingArmy.push(Viking);
+    }
+
+    addSaxon(Saxon){
+        this.saxonArmy.push(Saxon);
+    }
+    
+    vikingAttack(){
+        let saxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
+        let vikingIndex = Math.floor(Math.random() * this.vikingArmy.length);
+        let attackResult = this.saxonArmy[saxonIndex].receiveDamage(this.vikingArmy[vikingIndex].strength);
+        if (this.saxonArmy[saxonIndex].health <= 0) {
+            this.saxonArmy.pop(saxonIndex)
```

Comment:
Here, you should've used another array method like `.splice()`, just because the `.pop()` method removes the last element of the array. Even if you pass one index inside the method `.pop()`, it will always remove the last element of the array. It passes the tests because it's jasmine, but in another case it probably wouldn't work the way it is supposed to. 
The overall logic is correct, so, here's a method you could use: 
 ```
if (this.saxonArmy[saxonIndex].health <= 0) {
     this.saxonArmy.splice([saxonIndex], 1);
      }
```